### PR TITLE
Update machine controller to v0.9.4

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Update machine controller to v0.9.4

This will also remove the `containerRuntime` object from the api object we expose in the kubermatic-api.
The dashboard hardcoded `Docker` anyway - and we don't utilize it anywhere.
The Dashboard does not show the `containerRuntime` of the nodes at all - just the kubelet version.

**Release note**:
```release-note
Update machine-controller to v0.9.4
```

### CODE REVIEW

File: api/pkg/handler/node_v1.go [hunk 1]
```diff
@@ -237,18 +235,6 @@ func newCreateNodeForCluster(sshKeyProvider provider.NewSSHKeyProvider, projectP
 			return nil, errors.NewBadRequest("cannot create node without cloud provider")
 		}
 
-		// Support matrix: Ubuntu (crio + docker), containerlinux (docker), centos (docker)
```
Comment (alvaroaleman) [inline api/pkg/handler/node_v1.go:240]: Finally this goes away!

File: api/pkg/api/v2/node.go [hunk 2]
```diff
@@ -56,8 +56,7 @@ type OperatingSystemSpec struct {
 // NodeVersionInfo node version information
 // swagger:model NodeVersionInfoV2
 type NodeVersionInfo struct {
-	Kubelet          string                   `json:"kubelet"`
-	ContainerRuntime NodeContainerRuntimeInfo `json:"containerRuntime"`
```
Comment (p0lyn0mial) [inline api/pkg/api/v2/node.go:60]: I think that we should also remove the corresponding field from the dashboard, can you please file an issue fort that ?

**Comment by p0lyn0mial** [comment]
also the dashboard sends requests that contain `containerRuntime` field, have you checked that API will not explode when unmarshalling ?

File: api/pkg/api/v2/node.go (hunk 2 above)
Comment (alvaroaleman) [inline api/pkg/api/v2/node.go:60]: Did you check with the ui if this crates any issues? Probably fine as golang will just drop the extra attribute, right?

**Review by alvaroaleman — COMMENTED** [review]
Did you check the UI if this creates any issues?

File: api/pkg/api/v2/node.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/api/v2/node.go:60]: Issue for the dashboard: https://github.com/kubermatic/dashboard-v2/issues/807

Comment (mrIncompetent) [reply api/pkg/api/v2/node.go:60]: Issue for the UI: https://github.com/kubermatic/dashboard-v2/issues/807

**Comment by mrIncompetent** [comment]
Go will simply drop the field - so its safe to merge before the UI adapted

**Review by alvaroaleman — DISMISSED** [review]

**Comment by p0lyn0mial** [comment]
> Go will simply drop the field - so its safe to merge before the UI adapted

sgtm, this lgtm, not approving because you will have to fix the build.

**Comment by mrIncompetent** [comment]
@p0lyn0mial @alvaroaleman PTAL
I hat to fix the dep container for drone in master before & afterwards fix the vendor...

**Review by alvaroaleman — APPROVED** [review]

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**